src/osp/OSPApplication.cpp

```diff
@@ -30,3 +30,13 @@ OSPApplication::OSPApplication()
 {
 
 }
+
+void OSPApplication::debug_add_package(Package&& p)
```

Comment:
Careful with assuming that .emplace() will succeed.

1) Functions that just call a function on a member variable should either A) Return what the called function returns (Or a part of it, like the success / fail) or B) should assert at runtime that the inner call succeeded.
2) Do you really want to accept by rvalue-reference only? If so, that's fine, but idomatically most people assume you'll accept a copy as well as a move. If you need it to be rvalue-reference only, consider explicitly deleting the copy constructor and assignment operator off of the Package class.
3) There's a good chance that you want try_emplace over emplace, because try_emplace only actually move constructs the provided value if-and-only-if the insert operation will succeed.
4) If you don't want try_emplace, and you don't want to assert / return the bool, then maybe you want insert_or_assign to overwrite the existing key (if exists)

Reply:
We do indeed want move-only behavior, the Package class's copy constructor is already deleted (although the copy assign is missing, I'll delete that one too). I don't think it makes sense to allow the same key to be added multiple times so it should probably just fail. I'll assert for now; if @Capital-Asterisk thinks overwriting is better I can change it.